api/analyzers/source_analyzer.py

```diff
@@ -81,12 +81,11 @@ def first_pass(self, path: Path, ignore: list[str], graph: Graph) -> None:
             executor (concurrent.futures.Executor): The executor to run tasks concurrently.
         """
 
-        if any(path.rglob('*.java')):
-            analyzers[".java"].add_dependencies(path, self.files)
-        if any(path.rglob('*.py')):
-            analyzers[".py"].add_dependencies(path, self.files)
-
-        files = list(path.rglob('*.*'))
+        if any([file for file in files if file.suffix == ".java"]):
+            analyzers[".java"].add_dependencies(path, files)
+        if any([file for file in files if file.suffix == ".py"]):
+            analyzers[".py"].add_dependencies(path, files)
```

Comment:
Please maintain a list of supported extensions and use it instead of hardcoding the supported extensions.